torch/_logging/_internal.py

```diff
@@ -663,7 +667,15 @@ def format(self, record):
         return "\n".join(f"{prefix} {l}" for l in lines)
 
 
-DEFAULT_FORMATTER = TorchLogsFormatter()
+def _default_formatter():
+    fmt = os.environ.get(LOG_FORMAT_ENV_VAR, None)
+    if fmt is None:
+        return TorchLogsFormatter()
+    else:
+        return logging.Formatter(fmt)
```

Comment:
What if the formatter is invalid, what kind of error do we get?

Reply:
Currently, we throw a `ValueError: Formatting field not found in record` error. We could find a validate it but not sure how valuable/easy it would be.